src/main.ts

```diff
@@ -194,6 +199,42 @@ export function createEmitter<T extends Events>(options?: EmitterOptions) {
     }))
   }
 
+  function addGlobalHandler(globalEventHandler: GlobalEventHandler<T>, options?: EmitterOnOptions): () => void {
+    const { signal } = options ?? {}
+
+    if(signal?.aborted) {
+      return () => {}
+    }
+
+    globalHandlers.add(globalEventHandler)
+
+    signal?.addEventListener('abort', () => off(globalEventHandler))
+
+    return () => off(globalEventHandler)
```

Comment:
you _could_ create a const for this function and use for both

```suggestion
    const offHandler = () => off(globalEventHandler)

    signal?.addEventListener('abort', offHandler)

    return offHandler
```